packages/salesforce/src/Adaptor.js

```diff
@@ -363,16 +364,29 @@ export function bulkQuery(qs, options, callback) {
  * Create and execute a bulk job.
  * @public
  * @example
- * bulk('Patient__c', 'insert', { failOnError: true, pollInterval: 3000, pollTimeout: 240000 }, state => {
- *   return state.data.someArray.map(x => {
- *     return { 'Age__c': x.age, 'Name': x.name }
- *   })
- * });
+ * bulk(
+ *   "Patient__c",
+ *   "insert",
+ *   { failOnError: true, pollInterval: 3000, pollTimeout: 240000 },
+ *   (state) => state.someArray.map((x) => ({ Age__c: x.age, Name: x.name }))
+ * );
+ * @example <caption>Bulk upsert</caption>
+ * bulk(
+ *   "vera__Beneficiary__c",
+ *   "upsert",
+ *   { extIdField: "vera__External_ID__c", failOnError: true },
+ *   (state) => state.data
```

Comment:
There's nothing in this callback, what's the purpose of including it in this example?

I think this is a consistent problem in our examples - the inclusion of irrelevant or even misleading information.

We don't have to address this now, but for future reference a good set of examples would be `simple bulk insert`, `upsert with custom options`, `bulk operation with callback`, etc, where each has the minimum possible amount of syntax.

Reply:
Your right it's clear what is `state.data`, Is it an object, string or an array. 
I have updated the example to use an array of objects